# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.h
# Commit: 82fbedcdde75
# Full Hash: 82fbedcdde758a2ff7de93dabcd70675010822d1
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-30 03:48:30
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 3: Implement the algorithm for obtaining tls and use it for wasm signal handling. r=lth
#   
#   This is the third part of series of patches to Frame without tls pointer.
#   Here we preserve initial tls in all entry stubs and then use it to find a proper tls instance for a given frame.
#   
# ==============================================================================

diff -r 3e82e75c4992 -r 82fbedcdde75 js/src/jit/MacroAssembler.h
--- a/js/src/jit/MacroAssembler.h	Fri Oct 23 18:17:37 2020 +0000
+++ b/js/src/jit/MacroAssembler.h	Fri Oct 23 18:17:37 2020 +0000
@@ -4411,6 +4411,9 @@
       : ABIArgIterBase<VecT, WasmABIArgGenerator>(types) {}
 };
 
+wasm::TlsData* ExtractCalleeTlsFromFrameWithTls(wasm::Frame* fp);
+wasm::TlsData* ExtractCallerTlsFromFrameWithTls(wasm::Frame* fp);
+
 }  // namespace jit
 }  // namespace js
 